WordPress/gutenberg
# Promote fetchLinkSuggestions to non-experimental API w/out hard deprecation

## What?


Identical goal to #40052.
> This PR updates `__experimentalFetchLinkSuggestions` to `fetchLinkSuggestions` thereby making it non-experimental.

This PR adds a soft deprecation to prevent sites breaking in 3c21c52789116eb23d2e954dac6f6ed9ed533bc9 -- everything else is from @getdave's original pull request.


## Why?


The developer API has been stable for some time and treated as such by Gutenberg, WordPress Core and third-party plugin developers.

Moving straight to hard deprecation will break sites and goes against the WP policy of not breaking backward compatibility.

## How?


All renaming code is taken from #40052.

In addition it adds a deprecated.js file within the core-data package for maintaining `__experimentalFetchLinkSuggestions`.

I've listed the deprecated `since` tag as 6.1 as it appears deprecations use the WP version rather than the plugin version. Please correct me if this is incorrect.

## Testing Instructions





Per #40052, additionally:

* confirm `wp.coreData.__experimentalFetchLinkSuggestions()` is exposed in the console
* confirm running `wp.coreData.__experimentalFetchLinkSuggestions( 'sample' );` in the console throws a deprecation warning
* confirm running `wp.coreData.__experimentalFetchLinkSuggestions( 'sample' );` in the console returns the same results as the unprefixed function

## Screenshots or screencast
